code/game/objects/machinery/cloning/cloning.dm

```diff
@@ -0,0 +1,237 @@
+	if(.)
+		return
+	if(user.a_intent != INTENT_HELP)
+		return
+
+	if(!linked_machine)
+		// Try to find the machine nearby
+		linked_machine = locate() in range(1)
+		if(!linked_machine)
+			visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'connection not available'.</span>")
+			return TRUE
+
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps as its boots up and connects to \the [linked_machine]</span>")
+		return TRUE
+
+	if(linked_machine.occupant || linked_machine.timerid)
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'already processing clone'.</span>")
+		return TRUE
+
+	if(!linked_machine.beaker || linked_machine.beaker.reagents.total_volume < linked_machine.biomass_required)
+		visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> beeps in error, 'not enough biomass'.</span>")
+		return TRUE
+
+	visible_message("[icon2html(src, viewers(src))] <span><b>[src]</b> whirls as it starts to create a new clone.</span>")
+	linked_machine.grow_human()
+
+
+/obj/machinery/cloning/vats
+	name = "Clone Vat"
+	icon = 'icons/obj/machines/cloning.dmi'
+	icon_state = "cell_100"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 900
+
+	var/timerid
+	var/mob/living/carbon/human/occupant
+	var/obj/item/reagent_containers/glass/beaker
+
+	var/biomass_required = 40
+	var/grow_timer = 15 MINUTES
+
+
+/obj/machinery/cloning/vats/Initialize()
+	. = ..()
+	beaker = new /obj/item/reagent_containers/glass/beaker/biomass
+
+
+/obj/machinery/cloning/vats/Destroy()
+	deltimer(timerid)
+	timerid = null
+
+	// Force tthe clone out, if they have a client
+	if(occupant?.client)
+		eject_user()
+
+	QDEL_NULL(occupant)
+	QDEL_NULL(beaker)
+	return ..()
+
+
+/obj/machinery/cloning/vats/relaymove(mob/user)
+	eject_user()
+	return FALSE
+
+
+/obj/machinery/cloning/vats/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	if(user.a_intent == INTENT_HARM)
+		user.visible_message("<span class='notice'>[src] bangs on the glass</span>", "<span class='notice'>You bang on the glass</span>")
+		return TRUE
+
+	if(!beaker)
+		return
+
+	if(timerid || occupant) // You need to stop the process or remove the human first.
+		to_chat(user, "<span class='notice'>You can't get to the beaker while the machine growing a clone.</span>")
+		return
+
+	beaker.forceMove(drop_location())
+	user.put_in_hands(beaker)
+	beaker = null
+
+	update_icon()
+
+	return TRUE
+
+
+/obj/machinery/cloning/vats/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(.)
+		return
+
+	if(!ispath(I, /obj/item/reagent_containers/glass/beaker))
+		if(beaker)
+			to_chat(user, "<span class='warning'>A beaker is already loaded into the machine.</span>")
+			return
+
+		// Check if the beaker contains anything other than biomass juice
+		for(var/datum/reagent/R in I.reagents.reagent_list)
+			if(!istype(R, /datum/reagent/medicine/biomass) && !istype(R, /datum/reagent/medicine/biomass/xeno))
```

Comment:
@Rohesie this is where i wanted to use `ispath` but failed, because from docs i thought it would handle children too

Reply:
```cpp
var/datum/reagent/p = /datum/reagent/medicine/biomass
to_chat(world, "[ispath(p)]") //This shows 1
to_chat(world, "[istype(p)]") //This shows 0, it's not an instanced entity, it's just a path, similar to a string

p = new p() //Now we instance it and create the actual datum.
to_chat(world, "[ispath(p)]") //This shows 0, it's no longer a path, it's an instanced entity now.
to_chat(world, "[istype(p)]") //It has been instanced now with the correct type, so this outputs 1.
```

Reply:
Specifically i was looking to do the subpath matching
`ispath(/my/path/child, /my/path) // TRUE`

Ah i was confused and thought it was a list of paths (even though im using the instances)
I've fixed that by doing the ispath on the actual path, should fix my usage.

Reply:
Could you use more descriptive variable names than `I` and `R`?
Single-letter upper-case is widespread in use, but not a really good standard.